terminology.rst

```diff
@@ -206,3 +212,17 @@ Raiden Terminology
 
    expiration
         Specific block after which the lock in the :term:`locked transfer` expires
+
+   offchain message
+        A message that is only sent between Raiden nodes, with no information that could be
+        forwarded to a contract - for example, a ``Processed`` message.
+
+   onchain message
+        A message that is sent between Raiden nodes but could as well be forwarded to a contract.
+        Notice that a message does not have to be published on a contract to be considered an
+        onchain message - It only has to be of a type that allows sending it to a contract in
+        total, e. g. ``WithdrawRequest``.
+
+   envelope message
+        A message that partly consists of information which may be forwarded to a contract later,
+        such as a :term:`balance proof`. An example is the ``LockedTransfer`` message.
```

Comment:
We're using `pack_balance_proof` in `EnvelopeMessage._data_to_sign`, so if I'm not missing anything, it is *always* a balance proof.